python/cudf/cudf/core/column/column.py

```diff
@@ -846,6 +846,17 @@ def as_categorical_column(self, dtype, **kwargs):
             ordered = False
 
         sr = cudf.Series(self)
+
+        # Re-label self w.r.t. the provided categories
+        if is_categorical_dtype(dtype) and hasattr(dtype, "categories"):
```

Comment:
One minor change because `hasattr` checks are actually quite expensive:
```suggestion
        if isinstance(dtype, (cudf.CategoricalDtype, pd.CategoricalDtype)):
```